rpcs3/Emu/System.cpp

```diff
@@ -2007,10 +2008,46 @@ game_boot_result Emulator::Load(const std::string& title_id, bool is_disc_patch,
 				sys_log.error("Booting HG category outside of HDD0!");
 			}
 
-			g_fxo->init<main_ppu_module>();
+			const auto _main = g_fxo->init<main_ppu_module>();
 
 			if (ppu_load_exec(ppu_exec, false, m_path, DeserialManager()))
 			{
+				if (g_cfg.core.ppu_debug)
+				{
```

Comment:
Can't you put these spu and ppu debug prog saving sections into functions?
It's getting increasingly worse to read the general flow of these code sections

Reply:
Fixed.